icechunk/src/change_set.rs

```diff
@@ -44,8 +45,12 @@ impl ChangeSet {
         self.deleted_groups.iter()
     }
 
-    pub fn user_attributes_updated_nodes(&self) -> impl Iterator<Item = &NodeId> {
-        self.updated_attributes.keys()
+    pub fn updated_arrays(&self) -> impl Iterator<Item = &NodeId> {
```

Comment:
Is this really "nodes_with_updated_attributes"?

Reply:
```
self.updated_arrays.keys().chain(self.updated_groups.keys())
``` 

Why are we chaining `updated_groups` here?

Reply:
:facepalm: great catch ...

It shows the quality of our conflict testing